### PR TITLE
[CARBONDATA-1430] Resolved Split Partition Bug When Extra Space was Present in the NewList

### PR DESCRIPTION
**Pull request by nehabhardwaj01**
Split Partition Raises Exception if any extra space is given in the SplitList elements.
This PR is to resolve this issue.

### CODE REVIEW

**Comment by CarbonDataQA** [comment]
SDV Build Failed with Spark 2.1, Please check CI http://144.76.159.231:8080/job/ApacheSDVTests/307/

File: integration/spark-common/src/main/scala/org/apache/carbondata/spark/util/CommonUtil.scala [hunk 1]
```diff
@@ -347,7 +347,7 @@ object CommonUtil {
       sys.error("The size of new list must be smaller than original list, please check again!")
     }
     val tempList = newListInfo.mkString(",").split(",")
```
Comment (xuchuanyin) [inline integration/spark-common/src/main/scala/org/apache/carbondata/spark/util/CommonUtil.scala]: Is it necessary to do `list.mkString.split`?

Comment (nehabhardwaj01) [reply integration/spark-common/src/main/scala/org/apache/carbondata/spark/util/CommonUtil.scala]: Actually, the List on which these functions are being performed stores values as a List[String]

Suppose the list has the following values: ('Korea', 'China', '(Japan, US, Africa)' , 'UK')
Without applying any function the list elements will be :
list(0) => Korea
list(1) => China
list(2) => (Japan, US, Africa)		//This list of elements is stored like a string
list(3) => UK

But When we perform list.mkString(",").split(",")
The list now have every element as individual value
list(0) => Korea
list(1) => China
list(2) => (Japan
list(3) => US
list(4) => Africa)
list(5) => UK

Comment (xuchuanyin) [reply integration/spark-common/src/main/scala/org/apache/carbondata/spark/util/CommonUtil.scala]: thanks, learned from this... :+1:

**Comment by CarbonDataQA** [comment]
SDV Build Success with Spark 2.1, Please check CI http://144.76.159.231:8080/job/ApacheSDVTests/363/

**Comment by ravipesala** [comment]
SDV Build Success , Please check CI http://144.76.159.231:8080/job/ApacheSDVTests/511/

**Comment by sraghunandan** [comment]
please add a test case

**Comment by jackylk** [comment]
LGTM

**Comment by ravipesala** [comment]
SDV Build Fail , Please check CI http://144.76.159.231:8080/job/ApacheSDVTests/574/

**Comment by ravipesala** [comment]
SDV Build Fail , Please check CI http://144.76.159.231:8080/job/ApacheSDVTests/575/

**Pull request closed without merging**